plugins/outputs/sumologic/README.md

```diff
@@ -0,0 +1,68 @@
+# SumoLogic Output Plugin
+
+This plugin sends metrics to [Sumologic HTTP Source](https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source)
+in HTTP messages, encoded using one of the output data formats.
+Currently metrics can be sent using one of the following data formats, supported
+by Sumologic HTTP Source:
+
+  * `graphite` - for Content-Type of `application/vnd.sumologic.graphite`
+  * `carbon2` - for Content-Type of `application/vnd.sumologic.carbon2`
+  * `prometheus` - for Content-Type of `application/vnd.sumologic.prometheus`
+
+### Configuration:
+
+```toml
+# A plugin that can send metrics to Sumo Logic HTTP metric collector.
+[[outputs.sumologic]]
+  ## Unique URL generated for your HTTP Metrics Source.
+  ## This is the address to send metrics to.
+  # url = "https://events.sumologic.net/receiver/v1/http/<UniqueHTTPCollectorCode>"
+
+  ## Data format to be used for sending metrics.
+  ## This will set the "Content-Type" header accordingly.
+  ## Currently supported formats: 
+  ## * graphite - for Content-Type of application/vnd.sumologic.graphite
+  ## * carbon2 - for Content-Type of application/vnd.sumologic.carbon2
+  ## * prometheus - for Content-Type of application/vnd.sumologic.prometheus
+  ##
+  ## More information can be found at:
+  ## https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#content-type-headers-for-metrics
+  ##
+  ## NOTE:
+  ## When unset, telegraf will by default use the influx serializer which is currently unsupported
```

Comment:
Does telegraf provide any configuration validation on plugins? eg could we fail if a user has a format that is not one of the supported and when not specified can we assume carbon2 or fail? If we dont, then we will send the influx format as logs which is not a good user experience

Reply:
It's not really possible with the way it's designed now. So there is a switch case in https://github.com/SumoLogic/telegraf/blob/cd6a9e359d44aecb5d0373186f78340d69b6d2fd/plugins/outputs/sumologic/sumologic.go#L112-L122 on the serializer coming from the config file but we don't have control over it from the plugin itself and unfortunately the default serializer is being set as `influx` when the serializers' are being built https://github.com/SumoLogic/telegraf/blob/1b1382cabf427eedfa73a47320503237bca6736f/config/config.go#L1887-L1889

---

I'm kind of working around this as below, checking the config and then setting the error so that we exit with an error if anything else than the 3 supported serializers was chosen

https://github.com/SumoLogic/telegraf/blob/cd6a9e359d44aecb5d0373186f78340d69b6d2fd/plugins/outputs/sumologic/sumologic.go#L112-L122

Reply:
Got it ok that looks like the best we will get then.